modules/deser-lib/src/index.ts

```diff
@@ -0,0 +1,144 @@
+import { decodeFirstSync, encodeCanonical } from 'cbor';
+
+/** Return a string describing value as a type. */
+function getType(value): string {
+  if (Array.isArray(value)) {
+    const types = value.map(getType);
+    if (!types.slice(1).every((value) => value === types[0])) {
+      throw new Error('Array elements are not of the same type');
+    }
+    return JSON.stringify([types[0]]);
+  }
+  if (typeof value === 'object') {
+    const properties = Object.getOwnPropertyNames(value);
+    properties.sort();
+    return JSON.stringify(
+      properties.reduce((acc, name) => {
+        acc[name] = getType(value[name]);
+        return acc;
+      }, {})
+    );
+  }
+  if (typeof value === 'string') {
+    if (value.startsWith('0x')) {
+      return 'bytes';
+    }
+    return 'string';
+  }
+  return JSON.stringify(typeof value);
+}
+
+/** Compare two buffers for sorting. */
+function bufferCompare(a: Buffer, b: Buffer) {
+  let i = 0;
+  while (i < a.length && i < b.length && a[i] == b[i]) {
+    i++;
+  }
+  if (i === a.length && i === b.length) {
+    return 0;
+  }
+  if (i === a.length || i === b.length) {
+    return a.length - b.length;
+  }
+  return a[i] - b[i];
+}
+
+/** Compare two array elements for sorting. */
+function elementCompare(a: any, b: any) {
```

Comment:
Missing: 
- js doc
- input type (please do not use any, use unkown instead if you must)
- return type